.github/workflows/build.yml

```diff
@@ -30,185 +34,288 @@ jobs:
       run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-network
 
-    #
-    # Test network packages
-    #
+    # Build consensus packages
+
+    - name: ouroboros-consensus [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus
+
+    - name: ouroboros-consensus-mock [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-mock
+
+    - name: ouroboros-consensus-test [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-test
+
+    - name: ouroboros-consensus-mock-test [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-mock-test
+
+    - name: ouroboros-consensus-byron [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-byron
+
+    - name: ouroboros-consensus-byronspec [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-byronspec
+
+    - name: ouroboros-consensus-byron-test [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-byron-test
+
+    - name: ouroboros-consensus-shelley [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-shelley
+
+    - name: ouroboros-consensus-shelley-test [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-shelley-test
+
+    - name: ouroboros-consensus-cardano [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-cardano
+
+    - name: ouroboros-consensus-cardano-test [build]
+      if: matrix.subset == 'consensus'
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus-cardano-test
+
+    # Test Win32 specific packages
 
     - name: Win32-network [test]
       if: matrix.os == 'windows-latest'
       run: cabal --builddir="$CABAL_BUILDDIR" run Win32-network:test
 
+    # Test network packages
+
     - name: ntp-client [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run ntp-client:test
 
     - name: io-sim [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run io-sim:test
 
     - name: io-sim-classes [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run io-sim-classes:test
 
     - name: typed-protocols-examples [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run typed-protocols-examples:tests
 
     - name: network-mux [test]
+      if: matrix.subset == 'network'
       run: cabal --builddir="$CABAL_BUILDDIR" run network-mux:test
 
     # issue: #1818
```

Comment:
Would you elaborate a bit on this line comment?

The last GitHub comment on Issue #1818 is

> Fixed by PR #1627

PR #1627 has lots of comments, that I didn't read yet. Did it actually resolve #1818? EG I'd guess no, because of your line comment here.

Reply:
I don't know the history to #1818.  The comment already existed and I'm just preserving it.  @coot  Can you comment?

Reply:
Ah, I see, my mistake. Thanks for clarifying.

Reply:
I think this comment is stale and can be removed.  Early on we needed to filter out some tests, which does not happen any more.